src/video/wayland/SDL_waylanddyn.c

```diff
@@ -150,8 +151,20 @@ SDL_WAYLAND_LoadSymbols(void)
 #include "SDL_waylandsym.h"
 
         if (SDL_WAYLAND_HAVE_WAYLAND_CLIENT) {
-            /* all required symbols loaded. */
-            SDL_ClearError();
+            if (SDL_WAYLAND_HAVE_WAYLAND_CLIENT_1_20) {
```

Comment:
If you consider libdecor to be very important, then perhaps

```suggestion
            if (SDL_WAYLAND_HAVE_WAYLAND_CLIENT_1_20 && SDL_WAYLAND_HAVE_WAYLAND_LIBDECOR) {
```

or more directly

```suggestion
            if (SDL_WAYLAND_HAVE_WAYLAND_LIBDECOR) {
```

so that the default is Wayland if libdecor is available, or X11 otherwise?

Reply:
I'm hesitant - hypothetically a user running on sway or Plasma wouldn't need libdecor at all, so 1.18 + xdg-decoration qualifies as well.

Reply:
Hmm, yes. Maybe `SDL_WAYLAND_HAVE_WAYLAND_CLIENT_1_20 || SDL_WAYLAND_HAVE_WAYLAND_LIBDECOR` then?

Reply:
Note that #5436 will make `SDL_WAYLAND_HAVE_WAYLAND_CLIENT_1_20` not exist in some build configurations (`--disable-wayland-shared` + older Wayland at build-time), so this and/or #5436 will likely need some adjustment.

Reply:
> Note that #5436 will make `SDL_WAYLAND_HAVE_WAYLAND_CLIENT_1_20` not exist in some build configurations (`--disable-wayland-shared` + older Wayland at build-time), so this and/or #5436 will likely need some adjustment.

Actually, no, that shouldn't be a problem, because this part is `#ifdef SDL_VIDEO_DRIVER_WAYLAND_DYNAMIC` anyway.